### PR TITLE
No Bug: Add Log handler and debug menu for JitsiSDK logs.

### PR DESCRIPTION
**Pull request by iccub**
## Submitter Checklist:

- [ ] *Unit Tests* are updated to cover new or changed functionality
- [ ] User-facing strings use `NSLocalizableString()`
- [ ] New or updated UI has been tested across:
  - [ ] Light & dark mode
  - [ ] Different size classes (iPhone, landscape, iPad)
  - [ ] Different dynamic type sizes

## Screenshots:

Zrzut ekranu 2024-01-11 o 23 29 49

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: Sources/Brave/Frontend/Settings/Debug/BraveTalkLogsViewController.swift [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  private func getLogs() -> String {
+    do {
+      let store = try OSLogStore(scope: .currentProcessIdentifier)
+      
+      return try store
+        .getEntries()
+        .compactMap { $0 as? OSLogEntryLog }
+        .filter { $0.category == "BraveTalk" && $0.subsystem == Bundle.main.bundleIdentifier }
+        .map { "[\($0.date.formatted())] \($0.composedMessage)" }
+        .joined()
```
Comment (kylehickinson) [inline Sources/Brave/Frontend/Settings/Debug/BraveTalkLogsViewController.swift]: Can we join this with a new line (`.joined(separator: "\n")`) so its a bit easier to parse?

File: Sources/Brave/Frontend/Settings/Debug/BraveTalkLogsViewController.swift [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    
+    logsTextView.isEditable = false
+    view.addSubview(logsTextView)
+    logsTextView.snp.makeConstraints {
+      $0.edges.equalToSuperview()
+    }
+    
+    let formatter = DateFormatter()
+    formatter.dateStyle = .short
+    formatter.timeStyle = .long
```
Comment (kylehickinson) [inline Sources/Brave/Frontend/Settings/Debug/BraveTalkLogsViewController.swift]: This isn't used?

File: Sources/BraveTalk/BraveTalkJitsiCoordinator.swift [hunk 3]
```diff
@@ -17,7 +17,11 @@ import JitsiMeetSDK
     return true
   }
   
-  public init() {}
+  public init() {
+    if !AppConstants.buildChannel.isPublic {
+      JitsiMeetLogger.add(BraveTalkJitsiLogHandler())
```
Comment (kylehickinson) [inline Sources/BraveTalk/BraveTalkJitsiCoordinator.swift:22]: Is `JitsiMeetLogger` holding this class strongly I guess? We don't have to hold onto our own ref to keep it alive?

**Review by kylehickinson — CHANGES_REQUESTED** [review]

File: Sources/BraveTalk/BraveTalkJitsiCoordinator.swift (hunk 3 above)
Comment (iccub) [reply Sources/BraveTalk/BraveTalkJitsiCoordinator.swift:22]: Thats how I understand it. It uses `CocoaLumberJack` under the hood which I did not use in the past.
https://github.com/jitsi/jitsi-meet/blob/master/ios/sdk/src/JitsiMeetLogger.m#L30
Looks like it's added here to an array and the log class is a singleton https://github.com/CocoaLumberjack/CocoaLumberjack/blob/16815175e00cfa2a4d5c6adcb8c3395ea38b6bac/Sources/CocoaLumberjack/DDLog.m#L618

I can retain it in the jitsi coordinator class too if you think it's safer/better

Comment (kylehickinson) [reply Sources/BraveTalk/BraveTalkJitsiCoordinator.swift:22]: Sounds good, just wanted to make sure!

**Review by kylehickinson — APPROVED** [review]

**Pull request merged**